crates/turbopack-core/src/compile_time_info.rs

```diff
@@ -77,6 +77,7 @@ macro_rules! free_var_references {
 pub enum CompileTimeDefineValue {
     Bool(bool),
     String(String),
+    JSON(String),
```

Comment:
I'm not sure about this unstructured form... storing a serde::Value would allow us to do more static evaluation, like `typeof` and could replace bool and string too.

Reply:
yeah, I wanted to do that (actually I did implement it that way), but the `serialization = "auto_for_input"` requires this to implement `Hash` and `Ord`, but `serde_json::Value` doesn't implement either of those